app/scripts/controllers/permissions/index.js

```diff
@@ -371,7 +434,9 @@ export class PermissionsController {
 
     // update permitted accounts to ensure that accounts are returned
     // in the same order every time
-    this.updatePermittedAccounts(origin, newPermittedAccounts)
+    // await because the accounts may not have changed when this function
+    // returns even if the caller awaits
```

Comment:
This comment does quite make sense—if a promise is returned then we should always `await` it, no?

Reply:
I suppose in general, although can't we imagine `async` functions for which some callers care about what they return and others do not?

In my mind, calling an `async` function without awaiting it is to say: "Go do this work, I don't care when it's done or necessarily what the result is, just that you go do it. While you do that, I'm going to continue executing over here without being blocked by you."

In this case, not `await`ing essentially creates a race condition.

Reply:
Aight, I'm removing it.